src/main/scala/org/virtuslab/internal/load/compose/Composer.scala

```diff
@@ -0,0 +1,98 @@
+package org.virtuslab.internal.load.compose
+
+import org.virtuslab.internal.YamlError
+import org.virtuslab.internal.load.{YamlReader, StringYamlReader}
+import org.virtuslab.internal.load.parse.{Event, ParserImpl}
+import org.virtuslab.internal.load.compose.Node
+
+import scala.annotation.tailrec
+
+trait Composer:
+  def compose(yaml: String): Either[YamlError, Node]
+  def compose(reader: YamlReader): Either[YamlError, Node]
+
+object ComposerImpl extends Composer with NodeTransform:
+
+  type ComposeResult = (Either[YamlError, Node], List[Event])
+
+  override def compose(yaml: String): Either[YamlError, Node] = compose(StringYamlReader(yaml))
```

Comment:
I propose to change the name StringYamlReader to YamlReader. 
The name StringYamlReader it would make sense only if we supported binary serialization.

```suggestion
  override def compose(yaml: String): Either[YamlError, Node] = compose(YamlReader(yaml))
```

Reply:
What about trait `YamlReader`? After renaming there will be a name conflict between them.

Reply:
I'll leave it as it is because you changed that name already in your branch ;)